edit/ui.cattle.io.navlink.vue

```diff
@@ -0,0 +1,436 @@
+        this.targetName = value;
+        break;
+      }
+    },
+    /**
+     * Set URL type based on existing data
+     */
+    setUrlType() {
+      if (this.isURL) {
+        this.currentLinkType = LINK_TYPE_URL;
+      }
+      if (this.isService) {
+        this.currentLinkType = LINK_TYPE_SERVICE;
+      }
+    },
+    /**
+     * Initialize resource on creation
+     */
+    setDefaultValues() {
+      if (!this.value.spec) {
+        // Link to URL is set as default option from the data
+        this.$set(this.value, 'spec', { toURL: '' });
+      }
+      if (!this.value.metadata) {
+        this.$set(this.value, 'metadata', {});
+      }
+      if (!this.value.spec.target) {
+        this.$set(this.value.spec, 'target', LINK_TARGET_BLANK);
+      }
+    },
+    /**
+     * Set namespace and name from the selected service
+     * @param {label: string, id: string, name: string, namespace: string} service
+     */
+    setService(service) {
+      if (service) {
+        const { name, namespace } = service;
+
+        this.$set(this.value.spec.toService, 'name', name);
+        this.$set(this.value.spec.toService, 'namespace', namespace);
+      }
+    },
+    /**
+     * Set paired values of namespace and name for the service
+     */
+    setCurrentService() {
+      const name = this.value.spec.toService?.name;
+      const namespace = this.value.spec.toService?.namespace;
+
+      if (name && namespace) {
+        this.currentService = `${ namespace }/${ name }`;
+      }
+    },
+    /**
+     * Generate automatically kebab case for the displayed label
+     */
+    setName() {
+      this.$set(this.value.metadata, 'name', normalizeName(this.value.spec.label));
+    },
+    /**
+     * Get error chained validation based on existing label
+     * @param {string} label
+     */
+    getError(label) {
+      return this.$store.getters['i18n/t']('validation.required', { key: this.t(label) });
+    },
+    /**
+     * Verify all fields are fulfilled or display footer notification
+     */
+    validate() {
+      this.errors = [];
+
+      switch (this.currentLinkType) {
+      case LINK_TYPE_URL:
+        if (!this.value.spec.toURL) {
+          this.errors.push(this.getError('navLink.tabs.link.toURL.label'));
+        }
+        break;
+
+      case LINK_TYPE_SERVICE:
+        if (!this.value.spec.toService.name || !this.value.spec.toService.namespace) {
+          this.errors.push(this.getError(`navLink.tabs.link.toService.service.label`));
+        }
+
+        if (!this.value.spec.toService.scheme) {
+          this.errors.push(this.getError(`navLink.tabs.link.toService.scheme.label`));
+        }
+        break;
+
+        // no default
+      }
+
+      if (!this.value.metadata.name) {
+        this.errors.push(this.getError('navLink.name.label'));
+      }
+    },
+    /**
+     * Validate error presence by allowing or resetting submission process
+     * @param {*} done Form submission callback
+     */
+    finish(done) {
```

Comment:
Think this was to be removed and validation was going to occur in a before hook?